lib/googleauth/credentials.rb

```diff
@@ -40,10 +40,55 @@ module Auth
     class Credentials
       TOKEN_CREDENTIAL_URI = "https://oauth2.googleapis.com/token".freeze
       AUDIENCE = "https://oauth2.googleapis.com/token".freeze
-      SCOPE = [].freeze
-      PATH_ENV_VARS = [].freeze
-      JSON_ENV_VARS = [].freeze
-      DEFAULT_PATHS = [].freeze
+
+      def self.scope
+        if @scope.nil?
```

Comment:
Is the apparently minimal speed improvement of memoization here worth the risk of potential staleness?

Reply:
I don't know. Lemme try it without memoizing.

Reply:
Updated and removed the memoization. One thing that is lost without memoizing is if a user tries to modify the returned value when using the constants.

```ruby
class MyCredentials < Google::Auth::Credentials
  SCOPE = "one"
end

MyCredentials.scope #=> ["one"]

# Modifications using the method are lost
MyCredentials.scope << "two"
MyCredentials.scope #=> ["one"]

# Modifications using the constant work
MyCredentials::SCOPE = ["one", "two"]
MyCredentials.scope #=> ["one", "two"]
```

But, if users are using the method it should work fine:

```ruby
class MyCredentials < Google::Auth::Credentials
  self.scope = "one"
end

MyCredentials.scope #=> ["one"]

# Modifications using the method are kept
MyCredentials.scope << "two"
MyCredentials.scope #=> ["one", "two"]

# Modifications using the constant are lost
MyCredentials::SCOPE = ["three"]
MyCredentials.scope #=> ["one", "two"]
```

I can see an argument to keep the first example behavior, as it maintains backwards compatibility the most. But, I would mark the constants as deprecated, and when 1.0 is released remove support for the constants and switch to the second example behavior.